contracts/SpokePool.sol

```diff
@@ -1,16 +1,164 @@
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@uma/core/contracts/common/implementation/Testable.sol";
+import "@uma/core/contracts/common/implementation/Lockable.sol";
+import "@uma/core/contracts/common/implementation/MultiCaller.sol";
 
+/**
+ * @title SpokePool
+ * @notice Contract deployed on source and destination chains enabling depositors to transfer assets from source to 
+ * destination. Deposit orders are fulfilled by off-chain relayers who also interact with this contract. Deposited
+ * tokens are locked on the source chain and relayers send the recipient the desired token currency and amount
+ * on the destination chain. Locked source chain tokens are later sent over the canonical token bridge to L1.
+ */
+contract SpokePool is Testable, Lockable, MultiCaller {
+    using SafeERC20 for IERC20;
+
+    // Timestamp when contract was constructed. Relays cannot have a quote time before this.
+    uint64 public deploymentTime;
+
+    // Track the total number of deposits. Used as a unique identifier for deposits.
+    uint256 public numberOfDeposits;
+
+    // // Address of WETH on L1. If the deposited token maps to this L1 token then wrap ETH to WETH on the users behalf.
+    // address public l1Weth;
+
+    struct DestinationToken {
+        address token;
+        address spokePool;
+        uint256 chainId;
+        bool depositsEnabled;
+    }
+
+    // Whitelist of origin token to destination token mappings.
+    mapping(address => DestinationToken) public whitelistedDestinationTokens;
+
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+    event WhitelistToken(address originToken, address destinationToken, uint256 destinationChainId, address spokePool);
+    event DepositsEnabled(address originToken, bool depositsEnabled);
+    event FundsDeposited(
+        uint256 nonce,
+        uint256 destinationChainId,
+        address recipient,
+        address sender,
+        address originToken,
+        address destinationToken,
+        uint256 amount,
+        uint64 relayerFeePct,
+        uint64 quoteTimestamp
+    );
+
+    constructor(
+        // address _l1Weth,
+        address timerAddress
+    ) Testable(timerAddress) {
+        deploymentTime = uint64(getCurrentTime());
+    }
+
+    /****************************************
+     *               MODIFIERS              *
+     ****************************************/
+
+    modifier onlyIfDepositsEnabled(address originToken) {
+        require(whitelistedDestinationTokens[originToken].depositsEnabled, "Deposits disabled");
+        _;
+    }
+
+    /**************************************
+     *          ADMIN FUNCTIONS           *
+     **************************************/
+
+    /**
+     * @notice Whitelist an origin token <-> destination oken pair for bridging.
+     */
+    function _whitelistToken(
```

Comment:
as I mentioned above I dont think we need to just whitelist a token, we actually need to whitelist a _path_ a token can traverse. i.e Arbitrum->Optimism, with both the originToken and destinationToken for that given _path_